### PR TITLE
Update navicat-premium-essentials to 12.0.16

### PR DESCRIPTION
**Pull request by alreece45**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.
- [x] The commit message includes the cask’s name and version.

### CODE REVIEW

**Comment by alebcay** [comment]
Are you sure this isn't a downgrade?

File: Casks/navicat-premium-essentials.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 cask 'navicat-premium-essentials' do
-  version '12.0.15'
-  sha256 '4c7967abf3b14326796d5dfb5c0b6678633183e889f677d802f7f4fd0e0042e3'
+  version '12.0.6'
```
Comment (alebcay) [inline Casks/navicat-premium-essentials.rb]: Are you sure this isn't a downgrade?

Comment (alreece45) [reply Casks/navicat-premium-essentials.rb]: Should really be 2.0.16, will see about getting an updated version

**Comment by alreece45** [comment]
The version was incorrect before, it should be 12.0.16

**Pull request merged**